desktop/app/installer.js

```diff
@@ -1,65 +1,32 @@
 import {app} from 'electron'
-import {exec} from 'child_process'
-import path from 'path'
-import fs from 'fs'
-import os from 'os'
+import {appInstallerPath, appBundlePath} from './paths'
+import exec from './exec'
 
 import {runMode} from '../shared/constants/platform.native.desktop'
 
 export default callback => {
-  const appPath = app.getAppPath()
-  // For testing when running manually via npm start
-  // const appPath = '/Applications/Keybase.app/Contents/Resources/app/'
-  // For testing running from DMG
-  // const appPath = '/Volumes/Keybase/Keybase.app/Contents/Resources/app/'
-
-  const resourcesPath = path.resolve(appPath, '..')
-  const appBundlePath = path.resolve(appPath, '..', '..', '..')
-  const installerExec = path.resolve(resourcesPath, 'KeybaseInstaller.app', 'Contents', 'MacOS', 'Keybase')
-
-  const platform = os.platform()
-  if (platform !== 'darwin') {
-    callback(null)
+  const installerPath = appInstallerPath()
+  if (!installerPath) {
+    callback(new Error('No installer path'))
     return
   }
+  const bundlePath = appBundlePath()
+  if (!bundlePath) {
+    callback(new Error('No bundle path for installer'))
+    return
+  }
+  const args = ['--app-path=' + bundlePath, '--run-mode=' + runMode]
```

Comment:
Yeah we can't protect against that

On Thursday, June 30, 2016, Gabriel Handford <notifications@github.com>
wrote:

> In desktop/app/installer.js
> <https://github.com/keybase/client/pull/3261#discussion_r69224447>:
>
> >      return
> >    }
> > +  const bundlePath = appBundlePath()
> > +  if (!bundlePath) {
> > +    callback(new Error('No bundle path for installer'))
> > +    return
> > +  }
> > +  const args = ['--app-path=' + bundlePath, '--run-mode=' + runMode]
>
> Is a good point, but anything that could set and env variable for the app
> (via launchctl setenv) could execute any command as user anyway?
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/keybase/client/pull/3261/files/c4a423b504ecbdcafa3d20a611fa29ff10d61bed#r69224447>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAfhUwY7ktyLSaG0LFjoV-L7NyLUzCRPks5qREnxgaJpZM4JAsd->
> .
>